examples/integrations/oci/atp-reactive/src/main/java/io/helidon/examples/integrations/oci/atp/reactive/AtpService.java

```diff
@@ -69,30 +113,84 @@ private void generateWallet(ServerRequest req, ServerResponse res) {
-                    }
-                })
-                .build());
+                                   .connectionPool(() -> {
+                                       try {
+                                           return pds.getConnection();
+                                       } catch (SQLException e) {
+                                           LOGGER.log(Level.SEVERE, "Error while setting up new connection", e);
+                                           throw new IllegalStateException("Error while setting up new connection", e);
+                                       }
+                                   })
+                                   .build());
+    }
+
+    /**
+     * Returns SSLContext based on cwallet.sso in wallet.
+     *
+     * @return SSLContext
+     */
+    public SSLContext getSSLContext(byte[] walletContent) throws IllegalStateException {
+        SSLContext sslContext = null;
+        try (ZipInputStream zis = new ZipInputStream(new ByteArrayInputStream(walletContent))) {
+            ZipEntry entry = null;
+            while ((entry = zis.getNextEntry()) != null) {
+                if (entry.getName().equals("cwallet.sso")) {
+                    KeyStore keyStore = KeyStore.getInstance("SSO", new OraclePKIProvider());
+                    keyStore.load(zis, null);
+                    TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance("PKIX");
+                    KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance("PKIX");
+                    trustManagerFactory.init(keyStore);
+                    keyManagerFactory.init(keyStore, null);
+                    sslContext = SSLContext.getInstance("TLS");
+                    sslContext.init(keyManagerFactory.getKeyManagers(), trustManagerFactory.getTrustManagers(), null);
+                }
+                zis.closeEntry();
+            }
+        } catch (Exception e) {
+            throw new IllegalStateException("Error while getting SSLContext from wallet.", e);
+        }
+        return sslContext;
+    }
+
+    /**
+     * Returns JDBC URL with connection description for the given service based on tnsnames.ora in wallet.
+     *
+     * @param walletContent
+     * @param tnsNetServiceName
+     * @return String
+     */
+    public String getJdbcUrl(byte[] walletContent, String tnsNetServiceName) throws IllegalStateException {
+        String jdbcUrl = null;
+        try (ZipInputStream zis = new ZipInputStream(new ByteArrayInputStream(walletContent))) {
+            ZipEntry entry = null;
+            while ((entry = zis.getNextEntry()) != null) {
+                if (entry.getName().equals("tnsnames.ora")) {
+                    jdbcUrl = new String(zis.readAllBytes(), StandardCharsets.UTF_8)
+                            .replaceFirst(tnsNetServiceName + "\\s+=\\s+", "jdbc:oracle:thin:@")
+                            .replaceAll("\\n[^\\n]+", "");
```

Comment:
I have checked the content etc of tnsnames.ora and since we are downloading that in code, this was the way to get it esp. since we can't specify TNS_ADMIN i.e. file location for it using MemoryMapped files etc.

Reply:
Updated.